components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -93,6 +86,46 @@ void Nudging::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
     m_num_src_levs = scorpio::get_dimlen(m_static_vertical_pressure_file,"lev");
     scorpio::eam_pio_closefile(m_static_vertical_pressure_file);
   }
+
+  /* Check for consistency between nudging files, map file, and remapper */
+
+  // Number of columns globally
+  auto m_num_cols_global = m_grid->get_num_global_dofs(); 
+
+  // Get the information from the first nudging data file
+  scorpio::register_file(m_datafiles[0],scorpio::Read);
```

Comment:
FYI, you don't need to manually open/close the file for this operation. `get_dimlen` will automatically register/close the file if not yet open.

Reply:
I don't recall, I would have to check.